#VRML_SIM R2023b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/floors/protos/RectangleArena.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/apartment_structure/protos/Wall.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/Roughcast.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/Parquetry.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/floors/protos/Floor.proto"

WorldInfo {
}
Viewpoint {
  orientation -0.5773502691896257 0.5773502691896257 0.5773502691896257 2.0943245915045035
  position -0.00010831743706449024 -0.00023089771263453288 11.679950213487075
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 5 5
  floorAppearance Parquetry {
    type "light strip"
  }
}
DEF GOAL Floor {
  translation -1.625 2 0.01
  size 1 0.5
  appearance Parquetry {
    type "light strip"
    colorOverride 1 0 0
  }
}
Robot {
  translation 1 -1.75 0.00763687
  rotation 0.006654741795632369 -0.0004635847993957894 0.9999777495028415 3.14
  children [
    DEF cam Camera {
      translation 0.175 0 0.035
      rotation 0 1 0 1.57
      name "cam"
      boundingObject Box {
        size 0.03 0.03 0.03
      }
      physics Physics {
      }
    }
    DEF camera Camera {
      translation 0.175 0 0.1
      boundingObject Box {
        size 0.05 0.05 0.05
      }
      physics Physics {
      }
    }
    RangeFinder {
      translation 0.175 0 0.075
      name "rf0"
      boundingObject DEF rf Box {
        size 0.05 0.05 0.05
      }
      physics DEF rf Physics {
      }
      lens DEF rf0 Lens {
      }
    }
    DEF ds3 DistanceSensor {
      translation 0.175 -0.1 0.05
      name "ds3"
      boundingObject DEF ds Box {
        size 0.05 0.05 0.05
      }
      physics DEF ds Physics {
      }
      lookupTable [
        0 0 0
        0.1 1000 0
        0.2 400 0
        0.3 50 0
      ]
      type "sonar"
    }
    DEF ds4 DistanceSensor {
      translation 0.175 0.1 0.05
      name "ds4"
      boundingObject DEF ds Box {
        size 0.05 0.05 0.05
      }
      physics DEF ds Physics {
      }
      lookupTable [
        0 0 0
        0.1 1000 0
        0.2 400 0
        0.3 50 0
      ]
      type "sonar"
    }
    DEF ds0 DistanceSensor {
      translation 0.1 0.175 0.05
      rotation 0 0 1 1.57
      name "ds0"
      boundingObject DEF ds Box {
        size 0.05 0.05 0.05
      }
      physics Physics {
      }
      lookupTable [
        0 0 0
        0.1 1000 0
        0.2 400 0
        0.3 50 0
      ]
      type "sonar"
    }
    DEF ds1 DistanceSensor {
      translation 0.1 -0.175 0.05
      rotation 0 0 1 -1.57
      name "ds1"
      boundingObject DEF ds Box {
        size 0.05 0.05 0.05
      }
      physics Physics {
      }
      lookupTable [
        0 0 0
        0.1 1000 0
        0.2 400 0
        0.3 50 0
      ]
      type "sonar"
    }
    DEF ds1 DistanceSensor {
      translation 0.15 0 0.05
      name "ds2"
      boundingObject USE ds
      physics Physics {
      }
      lookupTable [
        0 0 0
        0.1 1000 0
        0.2 400 0
        0.3 50 0
      ]
      type "infra-red"
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 2160.2463946702665
        axis 0 1 0
        anchor -0.01 -0.172 0.05
      }
      device [
        RotationalMotor {
          name "motor_2"
        }
        DEF PosSen PositionSensor {
          name "PS_1"
        }
      ]
      endPoint Solid {
        translation -0.010055267915465373 -0.17200433814312865 0.05006698597148474
        rotation 0.7300646571518405 0.48320631869608277 0.48323622582536735 4.40288102763444
        children [
          DEF wheel_2 Shape {
            appearance PBRAppearance {
              baseColor 0 0 0
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.05
              radius 0.05
            }
          }
        ]
        name "solid(1)"
        boundingObject USE wheel_2
        physics Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 1637.0031767312082
        axis 0 1 0
        anchor -0.01 0.172 0.05
      }
      device [
        RotationalMotor {
          name "motor_1"
        }
        DEF PosSen PositionSensor {
          name "PS_2"
        }
      ]
      endPoint Solid {
        translation -0.009988166288882537 0.1719957392524234 0.05007080031202063
        rotation -0.7758699583322624 0.4463028661377508 -0.4459142960645265 4.462114856687627
        children [
          USE wheel_2
        ]
        boundingObject DEF wheel Shape {
          geometry Cylinder {
            height 0.05
            radius 0.05
          }
        }
        physics Physics {
        }
      }
    }
    DEF Body Pose {
      translation 0 0 0.0515
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.5 0.3 1
            emissiveColor 0.1 0 1
          }
          geometry Box {
            size 0.3 0.3 0.1
          }
        }
      ]
    }
  ]
  boundingObject USE Body
  physics Physics {
  }
  controller "Balls_In_Maze_End"
}
DEF Exit_Wall Wall {
  translation 0.375 2 0
  size 3.25 0.05 0.5
  appearance Roughcast {
    colorOverride 0.5 0.5 0.5
    textureTransform TextureTransform {
      scale 2.4 1
    }
  }
}
DEF Right_Wall Wall {
  translation 2 0 0
  rotation 0 0 1 1.57
  name "wall(1)"
  size 4 0.05 0.5
  appearance Roughcast {
    colorOverride 0.5 0.5 0.5
    textureTransform TextureTransform {
      scale 2.4 1
    }
  }
}
DEF Left_Wall Wall {
  translation -2 0 0
  rotation 0 0 1 1.57
  name "wall(2)"
  size 4 0.05 0.5
  appearance Roughcast {
    colorOverride 0.5 0.5 0.5
    textureTransform TextureTransform {
      scale 2.4 1
    }
  }
}
DEF Start_Wall Wall {
  translation -0.375 -2 0
  name "wall(3)"
  size 3.25 0.05 0.5
  appearance Roughcast {
    colorOverride 0.5 0.5 0.5
    textureTransform TextureTransform {
      scale 2.4 1
    }
  }
}
Wall {
  translation -1.25 -1.25 0
  name "wall(4)"
  size 1.5 0.05 0.5
}
Wall {
  translation -1.25 1 0
  name "wall(5)"
  size 1.5 0.05 0.5
}
Wall {
  translation 1.25 1.375 0
  rotation 0 0 1 1.57
  name "wall(6)"
  size 1.25 0.05 0.5
}
Wall {
  translation 1.125 -1.25 0
  name "wall(7)"
  size 1.75 0.05 0.5
}
Wall {
  translation 0.25 -0.5 0
  name "wall(8)"
  size 2 0.05 0.5
}
Wall {
  translation 0.75 1 0
  name "wall(9)"
  size 1 0.05 0.5
}
Wall {
  translation -0.875 0.25 0
  name "wall(10)"
  size 0.75 0.05 0.5
}
Wall {
  translation -1.25 0.75 0
  rotation 0 0 1 1.57
  name "wall(11)"
  size 1 0.05 0.5
}
Wall {
  translation -1.75 -0.5 0
  name "wall(12)"
  size 0.5 0.05 0.5
}
Wall {
  translation 1.25 0.5 0
  rotation 0 0 1 1.57
  name "wall(13)"
  size 0.5 0.05 0.5
}
Wall {
  translation 0.75 0.25 0
  name "wall(14)"
  size 1 0.05 0.5
}
Wall {
  translation 1.625 -2 0
  name "wall(15)"
  size 0.75 0.05 0.5
  appearance Roughcast {
    colorOverride 0 0 0
    textureTransform TextureTransform {
      scale 2.4 1
    }
  }
}
DEF RBall Solid {
  translation -1.625 0.875 0.05917698266528728
  children [
    DEF Ball Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
        emissiveColor 1 0 0
      }
      geometry Sphere {
        radius 0.05
      }
    }
  ]
  name "RBall"
  boundingObject USE Ball
  physics Physics {
  }
}
DEF BBall Solid {
  translation -1.875 -1 0.05917698266528728
  children [
    DEF Ball Shape {
      appearance PBRAppearance {
        baseColor 0 0 1
        emissiveColor 0 0 1
      }
      geometry Sphere {
        radius 0.05
      }
    }
  ]
  name "BBall"
  boundingObject USE Ball
  physics Physics {
  }
}
DEF GBall Solid {
  translation -1.75 -1.5 0.0590114
  rotation 0.8087479910006469 -0.038909442835673984 -0.58686688636405 -2.2611063929837325
  children [
    DEF Ball Shape {
      appearance PBRAppearance {
        baseColor 0 1 0
        emissiveColor 0 1 0
      }
      geometry Sphere {
        radius 0.05
      }
    }
  ]
  name "GBall"
  boundingObject USE Ball
  physics Physics {
  }
}
DEF YBall Solid {
  translation -1.125 0.625 0.05917698266528728
  children [
    DEF Ball Shape {
      appearance PBRAppearance {
        baseColor 1 1 0
        emissiveColor 1 1 0
      }
      geometry Sphere {
        radius 0.05
      }
    }
  ]
  name "YBall"
  boundingObject USE Ball
  physics Physics {
  }
}
